Unfortunately I get an error on not finding the ```requested version "7.0".``` without the patch.

```bash
error: builder for '/nix/store/1nvgppn02bsqg7gzsmyr2674cqmpzvld-veridian-0-unstable-2024-12-25.drv' failed with exit code 101;
       last 25 log lines:
       >   -- Detecting CXX compiler ABI info
       >   -- Detecting CXX compiler ABI info - done
       >   -- Check for working CXX compiler: /nix/store/888bkaqdpfpx72dd8bdc69qsqlgbhcvf-gcc-wrapper-13.3.0/bin/c++ - skipped
       >   -- Detecting CXX compile features
       >   -- Detecting CXX compile features - done
       >   -- Configuring incomplete, errors occurred!
       >
       >   --- stderr
       >   CMake Error at CMakeLists.txt:9 (find_package):
       >     Could not find a configuration file for package "slang" that is compatible
       >     with requested version "7.0".
       >
       >     The following configuration files were considered but not accepted:
       >
       >       /nix/store/22yi2v7ac6659g245fhlcnhgaw93vc1p-sv-lang-3.0/lib/cmake/slang/slangConfig.cmake, version: 3.0.0
       >
       >
       >
       >   thread 'main' panicked at /build/veridian-0-unstable-2024-12-25-vendor/cmake-0.1.50/src/lib.rs:1098:5:
       >
       >   command did not execute successfully, got: exit status: 1
       >
       >   build script failed, must exit now
       >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/1nvgppn02bsqg7gzsmyr2674cqmpzvld-veridian-0-unstable-2024-12-25.drv'.
error: 1 dependencies of derivation '/nix/store/744h0gijn351yqph0lrjkfd3372jdd1p-nix-shell-env.drv' failed to build
```


Should I take your changes and add the patches back in?

--- a/pkgs/by-name/ve/veridian/package.nix
+++ b/pkgs/by-name/ve/veridian/package.nix
@@ -1,0 +1,81 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+
+  cmake,
+  makeWrapper,
+  pkg-config,
+
+  boost,
+  fmt_11,
+  openssl,
+  sv-lang,
+  mimalloc,
+
+  verible,
+  verilator,
+}:
+rustPlatform.buildRustPackage {
+  pname = "veridian";
+  version = "0-unstable-2024-12-25";
+
+  src = fetchFromGitHub {
+    owner = "vivekmalneedi";
+    repo = "veridian";
+    rev = "d094c9d2fa9745b2c4430eef052478c64d5dd3b6";
+    hash = "sha256-3KjUunXTqdesvgDSeQMoXL0LRGsGQXZJGDt+xLWGovM=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-qJQD9HjSrrHdppbLNgLnXCycgzbmPePydZve3A8zGtU=";
+
+  buildFeatures = [ "slang" ];
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    cmake
+    makeWrapper
+    pkg-config
+  ];
+
+  buildInputs = [
+    boost
+    fmt_11
+    openssl
+    sv-lang
+    mimalloc
+  ];
+
+  # the tests also need these to be on the PATH
+  nativeCheckInputs = [
+    verible
+    verilator
+  ];
+
+  postInstall =
+    let
+      runtimePathDeps = [
+        verible
+        verilator
+      ];
+    in
+    ''
+      wrapProgram $out/bin/veridian \
+        --prefix PATH : ${lib.makeBinPath runtimePathDeps}
+    '';
+
+  env = {
+    OPENSSL_NO_VENDOR = "1";
+    RUSTFLAGS = "-C link-args=-lmimalloc";
+    # this is needed so that veridian doesn't try to build the sv-lang package itself
+    SLANG_INSTALL_PATH = sv-lang;
+  };
+
+  meta = {
+    description = "SystemVerilog Language Server";
+    homepage = "https://github.com/vivekmalneedi/veridian";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.hakan-demirli ];
+  };
+}

